src/lib.rs

```diff
@@ -19,56 +19,69 @@ pub struct Args {
     // Set defaults
-    let mut config = Config::default();
-    config
-        .set_default::<&str>("cfg_file", concat!(env!("CARGO_PKG_NAME"), ".toml"))
+    let mut builder = Config::builder();
+    builder = builder
+        .set_default("cfg_file", concat!(env!("CARGO_PKG_NAME"), ".toml"))
         .context(ErrorKind::ConfigImmutable)?;
-    config
-        .set_default::<&str>("Log.file", "_CONSOLE_")
+    builder = builder
+        .set_default("Log.file", "_CONSOLE_")
         .context(ErrorKind::ConfigImmutable)?;
-    config
-        .set_default::<&str>("Web.address", "127.0.0.1")
+    builder = builder
+        .set_default("Web.address", "127.0.0.1")
         .context(ErrorKind::ConfigImmutable)?;
-    config
+    builder = builder
         .set_default("Web.port", 7878)
         .context(ErrorKind::ConfigImmutable)?;
 
     // Determine config file
-    if let Ok(config_file) = env::var("AARDWOLF_CONFIG") {
-        config
-            .set("cfg_file", config_file)
-            .context(ErrorKind::ConfigImmutable)?;
-    }
+    let config_file = env::var("AARDWOLF_CONFIG").unwrap();
+    builder = builder
+        .set_override("cfg_file", config_file.clone())
+        .context(ErrorKind::ConfigImmutable)?;
 
-    if let Some(config_path) = parsed_args.config {
-        config
-            .set("cfg_file", config_path.to_str())
+    builder = builder.add_source(File::new(&config_file, FileFormat::Json)); // Or whatever format
+
+    let config_path = parsed_args.config.unwrap();
+        builder = builder
+            .set_override("cfg_file", config_path.to_str())
             .context(ErrorKind::ConfigImmutable)?;
+    //}
+
+    // Apply environmenet variable overrides
+    let env_vars = Environment::with_prefix("AARDWOLF")
+        .separator("_")
+        .ignore_empty(true);
+    builder = builder.add_source(env_vars);
+//
+    let log_path = parsed_args.log.unwrap(); 
+    builder = builder
+        .set_override("Log.file", log_path.to_str())
+        .context(ErrorKind::ConfigImmutable)?;
+    
+    //Err(ErrorKind::ConfigImmutable.into())
+
+    match builder.build() {
+        Ok(config) => {
+            env::set_var("DATABASE_URL", db_conn_string(&config)?);
+            return Ok(config)
+        },
+        Err(e) => {
+            // Throw an error
+            return Err(e.into());
+        }
     }
 
     // Merge config file and apply overrides
-    let config_file_string = config
-        .get_string("cfg_file")
-        .context(ErrorKind::ConfigMissingKeys)?;
-    let config_file = config::File::with_name(&config_file_string);
-    config.merge(config_file).context(ErrorKind::ConfigImmutable)?;
+    //let config_file_string = config
+    //    .get_string("cfg_file")
+    //    .context(ErrorKind::ConfigMissingKeys)?;
+    //let config_file = config::File::with_name(&config_file_string);
+    //config.merge(config_file).context(ErrorKind::ConfigImmutable)?;
```

Comment:
Question: Does the config builder not include a replacement for the merge functionality?  Or is there another reason you believe this functionality should be removed?

Reply:
config::builder deprecated merge.  It was one of the reasons why I was looking at the `config_manager` crate.

Reply:
Discussed in the matrix channel.  Turns out `add_source` covers this functionality, so this commented out code can just be removed.

Reply:
I'll be making some changes to this hopefully this weekend.

I'll make sure to take out the unwrap calls - I realized it wasn't a great solution when I wrote it, and submitted this PR without giving it the proper thought.